snooty/main.py

```diff
@@ -249,9 +249,9 @@ def _generate_build_identifiers(args: Dict[str, Optional[str]]) -> BuildIdentifi
     return identifiers
 
 
-def main() -> None:
+def main(argv: Optional[Sequence[str]] = None) -> None:
```

Comment:
Sorry, was snooping and just wondering: what's the point of adding this `argv` variable? Is there a way to set `fail_on_diagnostics` from the command line? If not, it seems like `argv` isn't actually used anywhere and this work is mostly a `ProjectConfig` change.

Reply:
oh, interesting. This was used when main was going to be invoked internally by the test, but it looks like it's now called via subprocess.